components/script/dom/bindings/js.rs

```diff
@@ -391,6 +392,54 @@ impl<T: DomObject> HeapSizeOf for MutNullableJS<T> {
     }
 }
 
+/// A holder that allows to lazily initialize the value only once
+/// `JS<T>`, using OnceCell
+/// Essentially a `OnceCell<JS<T>>`.
+///
+/// This should only be used as a field in other DOM objects; see warning
+/// on `JS<T>`.
+#[must_root]
+pub struct OnceCellJS<T: DomObject> {
+    ptr: OnceCell<JS<T>>,
+}
+
+impl<T: DomObject> OnceCellJS<T> {
+    /// Create a new `OnceCellJS` with value
+    pub fn new() -> OnceCellJS<T> {
```

Comment:
Make that the impl of `Default` for `OnceCellJS<T>` and this is good to go!

Thanks for keeping up with my comments!